lib/membrane/video_compositor/scene/object/layout.ex

```diff
@@ -10,11 +10,62 @@ defmodule Membrane.VideoCompositor.Scene.Object.Layout do
   """
   alias Membrane.VideoCompositor.Scene.{Object, Resolution}
 
+  defmodule RustlerFriendly do
+    @moduledoc false
+    # A rustler-friendly version of the Layout, prepared for rust serialization
+    alias Membrane.VideoCompositor.Scene.Object.RustlerFriendly, as: RFObject
+    alias Membrane.VideoCompositor.Scene.Resolution
+
+    @typedoc """
+    A rustler-friendly version of the internal name
+    """
+    @type internal_name :: RFObject.name()
+
+    @typedoc """
+    A rustler-friendly version of the inputs map. Keep in mind that this maps
+    __internal names__ => __scene object names__.
+    """
+    @type inputs :: %{internal_name() => RFObject.name()}
+    @type output_resolution :: {:resolution, Resolution.t()} | {:name, Object.name()}
+
+    # in a more 'final' product this should be some kind of a layout identifier.
+    # I thought of making this a UUID that would correspond to an implementation
+    # on the rust side, but layout names would work fine too.
+    @type rust_representation :: integer()
```

Comment:
Defining `rust_representation` `type` twice in `Layout` and `RFLayout` seems unnecessary. Moreover those types are inconsistent with each other since one is defined as `integer()` and the other on is defined as `non_neg_integer()`